go/tasks/plugins/array/k8s/task.go

```diff
@@ -111,32 +111,47 @@ func (t Task) Launch(ctx context.Context, tCtx core.TaskExecutionContext, kubeCl
 		return LaunchWaiting, nil
 	}
 
-	err = kubeClient.GetClient().Create(ctx, pod)
-	if err != nil && !k8serrors.IsAlreadyExists(err) {
-		if k8serrors.IsForbidden(err) {
-			if strings.Contains(err.Error(), "exceeded quota") {
-				// TODO: Quota errors are retried forever, it would be good to have support for backoff strategy.
-				logger.Infof(ctx, "Failed to launch  job, resource quota exceeded. Err: %v", err)
-				t.State = t.State.SetPhase(arrayCore.PhaseWaitingForResources, 0).SetReason("Not enough resources to launch job")
-			} else {
-				t.State = t.State.SetPhase(arrayCore.PhaseRetryableFailure, 0).SetReason("Failed to launch job.")
+	// Check for existing pods to prevent unnecessary Resource-Quota usage: https://github.com/kubernetes/kubernetes/issues/76787
+	existingPod := &corev1.Pod{}
+	err = kubeClient.GetClient().Get(context.Background(), client.ObjectKey{
```

Comment:
This is scary 😮 . It means we will issue two requests for every create. A couple of things:
1. Can you add a metric to count how many times we do a `Get` and find the pod?
2. Can you make it a config setting to control whether to do a Get before create or not? 
3. I think adding progress limit per step will greatly improve the reliability and success rate here... Is that something you are also working on? even if it's a config (e.g. launch 10 at a time) will make sure we will at most retry those 10 not 1000...

Reply:
I moved to use `cache` instead of k8s-client . Believe that will add a bit of delay but should resolve most of the concerns ?  

Also need to start working on 3 . Trying to get this in a good state otherwise and have a couple of use-cases running in prod first.